pkg/reconciler/pipelinerun/resources/resultrefresolution_test.go

```diff
@@ -554,3 +519,189 @@ func lessResolvedResultRefs(i, j *ResolvedResultRef) bool {
 	}
 	return strings.Compare(fromI, fromJ) < 0
 }
+
+func TestCheckMissingResultReferences(t *testing.T) {
+	for _, tt := range []struct {
+		name             string
+		pipelineRunState PipelineRunState
+		targets          PipelineRunState
+		wantErr          string
+	}{{
+		name:             "Valid: successful result references resolution - params",
+		pipelineRunState: pipelineRunState,
+		targets: PipelineRunState{
+			pipelineRunState[1],
```

Comment:
not related to this PR, I think it's hard to review this kind of test. Maybe one solution is to separate them into different vars with meaningful names?😂